backend/services/formatResearchData.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Function for converting research data.
+ * @param {Object} researchData - Research data
+ * @returns {string[]} Formatted research data
+ */
+function formatResearchData (researchData) {
+    let formattedResearchData = '';
+
+    for (const key in researchData[0])
+    {
+        formattedResearchData+=key + ',';
+    }
+
+    formattedResearchData+='\n';
+
+    let index = 0;
+    while (index < researchData.length){
+        let line = '';
+        for (const key in researchData[index])
+        {
+            line+=researchData[index][key] + ',';
```

Comment:
I think that whatever exists in `researchData[0]`, it will be added twice to the `formattedResearchData` string. First in lines 9-12 and then again here in this while loop when `index` is 0. (I am not sure if this is intentional)